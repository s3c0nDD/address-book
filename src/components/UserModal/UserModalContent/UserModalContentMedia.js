import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';

import CardMedia from '@material-ui/core/CardMedia';

const useStyles = makeStyles(theme => ({
  media: {
    height: 200,
    [theme.breakpoints.up('sm')]: {
      height: 260,
    },
    [theme.breakpoints.up('md')]: {
      height: 300,
    },
    [theme.breakpoints.up('lg')]: {
      height: 340,
    }
  }
}));

const UserModalContentMedia = ({ user }) => {
  const classes = useStyles();

  const photoTitle = useMemo(() => (
    `${user.name.title} ${user.name.first} ${user.name.last} photo`
  ), [user]);

  return (
    <CardMedia
      className={classes.media}
      image={user.picture.large}
      title={photoTitle}
    />
  );
};

export default UserModalContentMedia;

UserModalContentMedia.propTypes = {
  user: PropTypes.shape({
    name: PropTypes.shape({
      title: PropTypes.string.isRequired,
      first: PropTypes.string.isRequired,
      last: PropTypes.string.isRequired
    }).isRequired,
    picture: PropTypes.shape({
      large: PropTypes.string.isRequired
    }).isRequired,
  }).isRequired
};