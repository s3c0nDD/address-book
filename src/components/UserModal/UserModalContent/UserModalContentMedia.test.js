import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import userMockData from '../../mocks/userMock'
import UserModalContentMedia from './UserModalContentMedia'

describe('UserModalContentMedia', () => {
  let shallow;
  let wrapper;
  let userMock = { ...userMockData };

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    wrapper = shallow(
      <UserModalContentMedia
        user={userMock}
      />
    );
  });

  it('shallowly renders and matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});