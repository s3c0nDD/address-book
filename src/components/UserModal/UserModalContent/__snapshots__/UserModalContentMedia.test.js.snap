// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserModalContentMedia shallowly renders and matches snapshot 1`] = `
<WithStyles(ForwardRef(CardMedia))
  className="makeStyles-media-1"
  image="https://wwww.image.com/data/1"
  title="Mr John Doe photo"
/>
`;
